### PR TITLE
Address GRC Bugs

### PR DESCRIPTION
**Pull request by mprahl**
See each commit for details.

### CODE REVIEW

File: frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx [hunk 1]
```diff
@@ -31,66 +29,36 @@ import { DiffModal } from '../../components/DiffModal'
 export function PolicyTemplateDetails(
   props: Readonly<{
     clusterName: string
-    apiGroup: string
-    apiVersion: string
-    kind: string
-    templateName: string
-    setParentTemplate: Dispatch<any>
+    template: any
```
Comment (dhaiducek) [inline frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx:32]: It almost feels like `template` should be a type that contains the props that were removed, but that might be a refactor for another day (if it's valid).

File: frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetailsPage.tsx [hunk 2]
```diff
@@ -38,6 +44,56 @@ export function PolicyTemplateDetailsPage() {
...
+    }
+
+    if (addon.metadata.name !== 'governance-policy-framework') {
+      continue
+    }
+
+    if (addon.metadata.annotations?.['addon.open-cluster-management.io/hosting-cluster-name']) {
+      templateClusterName = addon.metadata.annotations['addon.open-cluster-management.io/hosting-cluster-name']
+      // open-cluster-management-agent-addon is the default namespace but it shouldn't be used for hosted mode.
+      templateNamespace = addon.spec.installNamespace || 'open-cluster-management-agent-addon'
```
Comment (dhaiducek) [inline frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetailsPage.tsx:64]: Per SonarCloud: https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&pullRequest=3542&id=open-cluster-management_console&open=AZACWMrNQMxzjpPQgioY

```suggestion
      templateNamespace = addon.spec.installNamespace ?? 'open-cluster-management-agent-addon'
```

**Review by dhaiducek — DISMISSED** [review]
/hold for more qualified reviewers

Seems sound to me. I have one musing and one bubbling up SonarCloud issue.

File: frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx (hunk 1 above)
Comment (mprahl) [reply frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx:32]: It's a good point but I'd rather leave that out for expediency of getting the bugs fixed with RC quickly approaching.

**Review by dhaiducek — APPROVED** [review]

**Pull request merged**